loader/src/sources/cdc/api.js

```diff
@@ -0,0 +1,282 @@
+}
+
+function error(message, context) {
+  console.error(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Error);
+}
+
+async function* queryState(state) {
+  const PAGE_SIZE = 5000;
+
+  let offset = 0;
+  while (true) {
+    try {
+      const response = await got({
+        url: `https://data.cdc.gov/resource/5jp2-pgaw.json?$limit=${PAGE_SIZE}&$offset=${offset}&loc_admin_state=${state}`,
+      });
+      const results = JSON.parse(response.body);
+
+      for (const result of results) {
+        yield result;
+      }
+
+      if (results.length < PAGE_SIZE) {
+        return;
+      } else {
+        offset += PAGE_SIZE;
+      }
+    } catch (e) {
+      error(`Error fetching CDC data`, e);
+      return;
+    }
+  }
+}
+
+function formatStore(storeItems) {
+  if (storeItems.length == 0) {
+    return null;
+  }
+
+  let result;
+  Sentry.withScope((scope) => {
+    const base = storeItems[0];
+    scope.setContext("location", {
+      id: base.provider_location_guid,
+      name: base.loc_name,
+      provider: "cdc",
+    });
+
+    const storeExternalId = getStoreExternalId(base);
+    if (!storeExternalId) {
+      return null;
+    }
+
+    const addressLines = [
+      base.loc_admin_street1,
+      base.loc_admin_street2,
+    ].filter(Boolean);
+
+    const metaFields = [
+      "insurance_accepted",
+      "walkins_accepted",
+
+      // TODO: better structure open hours information
+      "sunday_hours",
+      "monday_hours",
+      "tuesday_hours",
+      "wednesday_hours",
+      "thursday_hours",
+      "friday_hours",
+      "saturday_hours",
+    ];
+    const meta = {};
+    for (const field of metaFields) {
+      meta[field] = base[field];
+    }
+
+    const productFields = [
+      "ndc",
+      "med_name",
+      "in_stock",
+      "supply_level",
+      "quantity_last_updated",
+    ];
+    const productList = storeItems.map((item) => {
+      const out = {};
+      for (const field of productFields) {
+        out[field] = item[field];
+      }
+      return out;
+    });
+
+    result = {
+      id: `cdc:${base.provider_location_guid}`,
+      external_ids: [
+        ["vaccines_gov", base.provider_location_guid],
+        storeExternalId,
+      ],
+      name: titleCase(base.loc_name),
+      provider: "cdc",
+
+      address_lines: addressLines,
+      city: base.loc_admin_city,
+      state: base.loc_admin_state,
+      postal_code: base.loc_admin_zip,
+      position: {
+        longitude: base.longitude,
+        latitude: base.latitude,
+      },
+      info_phone: base.loc_phone,
+      info_url: base.web_address,
+      meta,
+
+      availability: {
+        source: "cdc",
+        checked_at: new Date().toISOString(),
+        valid_at: formatValidAt(productList),
+        available_count: getAvailableCount(productList),
+        available: formatAvailable(productList),
+        capacity: formatCapacity(productList),
```

Comment:
I’m not really sure we should set the `capacity` field… since this isn’t really appointments, we shouldn’t try to guess something here. Better to just set the top level `available` and `available_count` fields, I think.